nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/SetPropertiesAction.java

```diff
@@ -64,7 +71,20 @@ public SetPropertyComputation() {
 
         @Override
         protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            // prevent unmodifiabble map implementation
+            properties = new HashMap<>(properties);
```

Comment:
all this param peparation can be done on the startBucket method, this will make the computation faster and easier to read